frontend/src/components/header.js

```diff
@@ -42,65 +50,63 @@ BootstrapNavLink.propTypes = {
-                    ))}
-                >
-                    {routes
-                        .filter((route) => !route.hidden)
-                        .map((route) => (
-                            <NavDropdown.Item
-                                key={route.route}
-                                as="span"
-                                tabIndex="0"
+        <Tab.Container activeKey={key} onSelect={(k) => setKey(k)}>
+            <Row className="justify-content-between right-padding">
+                <Col auto>
+                    <Nav className="flex-row navbar-tabs" variant="tabs">
+                        {routes.map((route) => (
+                            <BootstrapNavItem
+                                eventKey={route.route}
+                                to={route.route}
                             >
-                                <BootstrapNavLink
-                                    to={route.route}
-                                    title={route.description}
-                                >
-                                    {route.name}
-                                </BootstrapNavLink>
-                            </NavDropdown.Item>
+                                {route.name}
+                            </BootstrapNavItem>
                         ))}
-                </NavDropdown>
-            </Navbar.Brand>
-            <Nav className="mr-auto">
-                {routes.map((route) => (
-                    <Route path={route.route} key={route.route}>
-                        {(route.subroutes || []).map((subroute) => {
-                            const fullroute = `${route.route}${subroute.route}`;
-                            return (
-                                <BootstrapNavLink
-                                    to={fullroute}
-                                    key={fullroute}
-                                    title={subroute.description}
-                                >
-                                    {subroute.name}
-                                </BootstrapNavLink>
-                            );
-                        })}
-                    </Route>
-                ))}
-            </Nav>
-            {infoComponents.map((component, index) => (
-                <div key={index}>{component}</div>
-            ))}
-        </Navbar>
+                    </Nav>
+                </Col>
+                <Col md={"auto"}>
+                    <Row class>
+                        {infoComponents.map((component, index) => (
+                            <div key={index}>{component}</div>
+                        ))}
+                    </Row>
+                </Col>
+            </Row>
+            <Row className="navbar-subtabs">
+                <Tab.Content>
+                    <Nav variant="pills">
+                        {routes
+                            .filter((route) => route.route === key)
+                            .map((route) =>
```

Comment:
I would rather this logic be done before the return statement, and then just included here. E.g.,

```
const activeTabs = routes.filter ...;

...

<Nav>
   {activeTabs}
</Nav>
```

Reply:
Sure, makes sense to me.